### PR TITLE
Delete explainer feature

### PR DESCRIPTION
**Pull request by marialivia16**
This will delete an atom explainer only if the atom is taken down or draft. It works in 2 steps: first you click on the icon and you get a button and then you confirm the delete. The page will reload after the action is completed.

n67xtyxzm2

@philmcmahon

### CODE REVIEW

File: explainer-server/app/controllers/ExplainerApiImpl.scala [hunk 1]
```diff
@@ -102,6 +102,18 @@ class ExplainerApiImpl(
     })
   }
 
+  override def delete(id: String): Future[Unit] = {
+    getStatus(id).map( publicationStatus => {
+      if (publicationStatus == Draft || publicationStatus == TakenDown) {
+        explainerDB.load(id).map( explainer => {
+          explainerDB.delete(id)
+          sendKinesisEvent(explainer, s"Sending delete event for explainer $id to PREVIEW kinesis.", previewAtomPublisher, EventType.Takedown)
```
Comment (marialivia16) [inline explainer-server/app/controllers/ExplainerApiImpl.scala:110]: @philmcmahon There is no EventType delete in the thrift model (https://github.com/guardian/content-atom/blob/master/thrift/src/main/thrift/contentatom.thrift#L69). Do we need to add one?

Comment (philmcmahon) [reply explainer-server/app/controllers/ExplainerApiImpl.scala:110]: As far as capi is concerned 'Takedown' is the same as delete - results in the item being deleted from elasticsearch.

**Review by philmcmahon — COMMENTED** [review]
All looks great! My only thought was maybe that the confirm delete button should be red rather than green?

File: explainer-server/app/controllers/ExplainerApiImpl.scala (hunk 1 above)
Comment (marialivia16) [reply explainer-server/app/controllers/ExplainerApiImpl.scala:110]: Ok, cool. Thanks!

**Comment by philmcmahon** [comment]
👍 🗑

**Comment by marialivia16** [comment]
@philmcmahon Good point about the colour of the button. It is actually red, but because I'm hovering it looks green. But thinking about it now, I might as well remove the hover colour because when I hover out the button disappears and we'll never see the red.

**Pull request merged**